### ISSUE TITLE
Some collimation particle arrays have the wrong size

### ISSUE THREAD

**Issue by kyrsjo** [SixTrack/SixTrack#223]
When compiling with `./cmake_six HUGENPART COLLIMAT -CRLIBM STF DATAMODS` and gfortran 6.3.1, one recieves the following warning:

```
/home/kyrsjo/code/SixTrack/SixTrack/SixTrack_cmakesix_COLLIMAT_DATAMODS_HUGENPART_STF_NOCRLIBM_defaultcompiler_defaultbuildtype/track.f:6632:46:

      &              rcp, rcs, napx, enom_gev, part_hit, part_abs,       &
                                              1
Warning: Actual argument contains too few elements for dummy argument ‘lhit’ (20000/65536) at (1)
/home/kyrsjo/code/SixTrack/SixTrack/SixTrack_cmakesix_COLLIMAT_DATAMODS_HUGENPART_STF_NOCRLIBM_defaultcompiler_defaultbuildtype/track.f:6905:26:

      &                    part_hit, part_abs, part_impact, part_indiv,  &
                          1
Warning: Actual argument contains too few elements for dummy argument ‘lhit’ (20000/65536) at (1)
/home/kyrsjo/code/SixTrack/SixTrack/SixTrack_cmakesix_COLLIMAT_DATAMODS_HUGENPART_STF_NOCRLIBM_defaultcompiler_defaultbuildtype/track.f:6927:49:

      &                 rcp, rcs, napx, enom_gev, part_hit, part_abs,    &
                                                 1
Warning: Actual argument contains too few elements for dummy argument ‘lhit’ (20000/65536) at (1)
```
This happens in the calls to `collimate2` and `collimaterhic`.

This is the argument `part_hit` is of size maxn (which is 20k), while it is recieved in a buffer "lhit" inside the called functions. This buffer is of size "npart", which is normally 64, but here 65k (should probably be automatic, since we anyway pass by reference?).

The size "npart" is probably more correct, since the `part_hit` array is always treated as a tracking array, not a collimation "backend" particle array. However, the entire array is at some point initialized.

Since only the commonly-agreed-upon parts of the arrays are anyway used, I don't think there is any effect on results except an ugly warning.

**Comment by kyrsjo** [comment]
Same warning in gfortran 5.3.1.

**Comment by kyrsjo** [comment]
Affected variables in dbcommon:
part_hit, part_abs, part_select, nabs_type, part_impact, counted_r, counted_x, counted_y, secondary, tertiary, other, part_hit_before, part_indic, part_linteract, nabs_type

In dbcollim: nabs_type (this is ALSO defined in dbcommon, and is there part of the common block "stats".

Also, the "other" array seems to be missing from the particle compaction after a loss...

**Comment by kyrsjo** [comment]
And then we have the arrays "x00, xp00, y00, yp00" which are totally unused and could probably be deleted...

**Comment by kyrsjo** [comment]
Also `flagsec` in `collimate2`!

**Comment by kyrsjo** [comment]
Fixes in PR #226...

**Comment by kyrsjo** [comment]
Fixed in #226 .
